### PR TITLE
Global type size

### PR DESCRIPTION
**Pull request by localghost443**

### CODE REVIEW

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue [hunk 1]
```diff
@@ -276,21 +276,23 @@ export default {
...
     @include apos-list-reset();
+    margin-bottom: $spacing-double;
+  }
+
+  .apos-media-editor__detail {
+    @include typeMeta;
     color: var(--a-base-4);
-    font-size: map-get($font-sizes, default);
-    font-weight: 500;
-    margin-bottom: 20px;
```
Comment (stuartromanek) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:287]: where'd this margin go if `typeMeta` doesn't include it

File: modules/@apostrophecms/modal/ui/apos/components/AposModal.vue [hunk 2]
```diff
@@ -23,7 +23,7 @@
               <div v-if="hasSecondaryControls" class="apos-modal__controls--secondary">
                 <slot name="secondaryControls" />
               </div>
-              <h2 :id="id" class="apos-modal__heading o-heading">
+              <h2 :id="id" class="apos-modal__heading">
```
Comment (stuartromanek) [inline modules/@apostrophecms/modal/ui/apos/components/AposModal.vue:26]: `:)`

File: modules/@apostrophecms/notification/ui/apos/components/TheAposNotifications.vue [hunk 3]
```diff
@@ -127,5 +127,10 @@ export default {
 <style lang="scss" scoped>
   .apos-notifications {
     z-index: $z-index-notifications;
+    position: absolute;
+    right: 20px;
+    bottom: 20px;
+    display: flex;
+    flex-direction: column;
```
Comment (stuartromanek) [inline modules/@apostrophecms/notification/ui/apos/components/TheAposNotifications.vue:134]: back this out, different PR

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 4]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
   --a-primary-button-disabled: #{mix(white, $color, 40%)};
   // stylelint-enable
 }
+
+//Application Styles
+
+@mixin typeSmall {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:28]: kebab case for consistency

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 5]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+
+@mixin typeBase {
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-base);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-base);
+}
+
+@mixin typeTitle {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:44]: kebab

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 6]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+  margin: 0;
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-heading);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-tall);
+}
+
+@mixin typeDisplay {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:53]: kebab

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 7]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+  margin: 0;
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-display);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-base);
+}
+
+@mixin typeAffordance {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:62]: kebab .. Also this name is vague when seeing it in code .. is there something more concrete

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 8]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+  font-size: var(--a-type-small);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-base);
+  text-transform: uppercase;
+}
+
+//Novella Styles
+
+@mixin typeHelp {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:73]: kebab

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 9]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-small);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-tallest);
+}
+
+@mixin typeLarge {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:81]: kebab

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 10]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-large);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-tall);
+}
+
+@mixin typeMeta {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:89]: kebab

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 11]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-base);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-tallest);
+}
+
+@mixin typeBody {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:97]: kebab

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 12]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-large);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-tallest);
+}
+
+@mixin typeHeading {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:105]: kebab

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss [hunk 13]
```diff
@@ -49,3 +22,90 @@ $input-color-disabled: var(--a-base-4);
...
+
+@mixin typeSmall {
+  font-family: var(--a-family-default);
+  font-size: var(--a-type-small);
+  font-weight: var(--a-weight-base);
+  letter-spacing: var(--a-letter-base);
+  line-height: var(--a-line-tall);
+}
+
+@mixin typeBase {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss:36]: kebab

**Review by stuartromanek — CHANGES_REQUESTED** [review]
- remove the `apos-p-reset` mixin completely? 
- revert steamroll formatting on TheAposLogin.vue
- kebab mixins (or normalize the others, i defer to @abea on style)

Questions
- I don't understand the big move of button styles? It seems like that would be a pattern everything should follow
- Should `color` be part of the type mixins, since we have a dynamic primary text variable?

File: modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue [hunk 14]
```diff
@@ -32,7 +32,7 @@
               <span v-if="maxLabel" :class="maxError ? 'apos-modal-array-max-error' : ''">
                 {{ maxLabel }}
               </span>
-            </p>
```
Comment (localghost443) [inline modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue:35]: This is not technically speaking supposed to be a <p>.

File: modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue [hunk 15]
```diff
@@ -388,28 +388,22 @@ export default {
...
+    margin: $spacing-base;
   }
 
   .apos-modal-array-items__heading {
     position: relative;
-    margin: 20px 4px;
-  }
-
-  // Specificity needed due to AposButton rules
-  .apos-modal-array-items__add.apos-button {
```
Comment (localghost443) [inline modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue:400]: There was no reason for this to be position absolute.  Fixed while applying type

File: modules/@apostrophecms/schema/ui/apos/components/AposSchema.vue [hunk 16]
```diff
@@ -209,8 +209,7 @@ export default {
 <style lang="scss" scoped>
   .apos-field {
     .apos-schema /deep/ & {
-      margin-bottom: 30px;
-      letter-spacing: 0.5px;
+      margin-bottom: $spacing-triple;
```
Comment (localghost443) [inline modules/@apostrophecms/schema/ui/apos/components/AposSchema.vue:212]: There are instances where we weren't applying the variables we set up.

File: modules/@apostrophecms/ui/ui/apos/components/AposButton.vue [hunk 17]
```diff
@@ -145,374 +145,3 @@ export default {
   }
 };
 </script>
-
-<style lang="scss" scoped>
```
Comment (localghost443) [inline modules/@apostrophecms/ui/ui/apos/components/AposButton.vue:149]: This was all moved to _button.scss

File: modules/@apostrophecms/ui/ui/apos/scss/global/_theme.scss [hunk 18]
```diff
@@ -1,9 +1,3 @@
-body {
```
Comment (localghost443) [inline modules/@apostrophecms/ui/ui/apos/scss/global/_theme.scss:1]: This was the primary culprit of this entire PR. This was the only place we were defining the UI type, and it had the side effect of effecting project-cod

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue (hunk 1 above)
Comment (localghost443) [reply modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:287]: novella-meta (the only meta after switching to kebob), does not reset the margin

File: modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss (hunk 7 above)
Comment (localghost443) [reply modules/@apostrophecms/ui/ui/apos/scss/mixins/_theme_mixins.scss]: kebabed, and renamed "type help"

**Review by localghost443 — COMMENTED** [review]
- The move to _button.scss is consistent with _input.scss, it avoids overrides in 4 separate scoped areas. 
- color is not part of the type styles since the typography doesn't change between light/dark mode
- apos-p-reset is still sometimes useful
-

File: modules/@apostrophecms/ui/ui/apos/components/AposEmptyState.vue [hunk 19]
```diff
@@ -32,15 +32,15 @@ export default {
   }
 
   .apos-title {
+    @include type-title;
     margin: 0 0 10px;
   }
 
   .apos-hint {
```
Comment (localghost443) [inline modules/@apostrophecms/ui/ui/apos/components/AposEmptyState.vue:39]: In the designs, we're actually using italic.

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue [hunk 20]
```diff
@@ -1,15 +1,15 @@
 <template>
   <div class="apos-media-manager-selections">
     <div v-if="items.length">
-      <p class="apos-media-manager-selections__heading">
+      <div class="apos-media-manager-selections__heading">
         {{ items.length }} items selected
```
Comment (localghost443) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue:5]: should probably be wrapped in a span

File: modules/@apostrophecms/modal/ui/apos/components/AposModal.vue (hunk 2 above)
Comment (abea) [reply modules/@apostrophecms/modal/ui/apos/components/AposModal.vue:26]: ⚡

File: modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue [hunk 21]
```diff
@@ -234,25 +212,28 @@ export default {
...
-        text-decoration-line: underline;
+      button {
+        margin-top: $spacing-double;
       }
+    }
 
-      button {
-        font-size: map-get($font-sizes, input-label);
-        letter-spacing: 0.5px;
+    &__loader {
```
Comment (abea) [inline modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue:220]: I don't see this __loader class actually used anywhere.

Comment (abea) [reply modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue:220]: Yeah, it doesn't seem to be in here.

**Review by abea — APPROVED** [review]
A couple of things below to follow-up on later, plus the button style stuff we talked about.

The pager buttons aren't the same size anymore:
Screen Shot 2020-10-29 at 3 53 50 PM

This help text looks pretty tight to me (line spacing):
Screen Shot 2020-10-29 at 3 57 26 PM

File: modules/@apostrophecms/modal/ui/apos/components/ApostropheModal.vue [hunk 22]
```diff
@@ -63,7 +63,6 @@ export default {
   border-radius: 2px;
   box-shadow: 0 2px 8px rgba(0, 0, 0, 0.33);
   transition: all 0.3s ease;
-  font-family: map-get($font-families, default);
```
Comment (abea) [inline modules/@apostrophecms/modal/ui/apos/components/ApostropheModal.vue:66]: This whole file was deleted, so you'll have a little merge conflict. Pretty easy to resolve.

**Review by abea — APPROVED** [review]
#bebetter

**Pull request merged**